napari/layers/points/points.py

```diff
@@ -1979,110 +1911,515 @@ def _set_drag_start(
+            feature_defaults=feature_defaults,
+            properties=properties,
+            text=text,
+            symbol=symbol,
+            size=size,
+            border_width=border_width,
+            border_width_is_relative=border_width_is_relative,
+            border_color=border_color,
+            border_color_cycle=border_color_cycle,
+            border_colormap=border_colormap,
+            border_contrast_limits=border_contrast_limits,
+            face_color=face_color,
+            face_color_cycle=face_color_cycle,
+            face_colormap=face_colormap,
+            face_contrast_limits=face_contrast_limits,
+            out_of_slice_display=out_of_slice_display,
+            n_dimensional=n_dimensional,
+            name=name,
+            metadata=metadata,
+            scale=scale,
+            translate=translate,
+            rotate=rotate,
+            shear=shear,
+            affine=affine,
+            opacity=opacity,
+            blending=blending,
+            visible=visible,
+            cache=cache,
+            property_choices=property_choices,
+            experimental_clipping_planes=experimental_clipping_planes,
+            shading=shading,
+            canvas_size_limits=canvas_size_limits,
+            antialiasing=antialiasing,
+            shown=shown,
+        )
+
+        self.events.add(
+            edge_width=deprecation_warning_event(
+                "layer.events", "edge_width", "border_width", "0.6.0"
+            ),
+            current_edge_width=deprecation_warning_event(
+                "layer.events",
+                "current_edge_width",
+                "current_border_width",
+                "0.6.0",
+            ),
+            edge_width_is_relative=deprecation_warning_event(
+                "layer.events",
+                "edge_width_is_relative",
+                "border_width_is_relative",
+                "0.6.0",
+            ),
+            edge_color=deprecation_warning_event(
+                "layer.events", "edge_color", "border_color", "0.6.0"
+            ),
+            current_edge_color=deprecation_warning_event(
+                "layer.events",
+                "current_edge_color",
+                "current_border_color",
+                "0.6.0",
+            ),
+        )
+
+    @property
+    def _points_data(self) -> np.ndarray:
+        """Spatialy distributed coordinates."""
+        return self.data
+
+    @property
+    def data(self) -> np.ndarray:
+        """(N, D) array: coordinates for N points in D dimensions."""
+        return self._data
+
+    @data.setter
+    def data(self, data: Optional[np.ndarray]):
+        data, _ = fix_data_points(data, self.ndim)
+        cur_npoints = len(self._data)
+        self._data = data
+
+        # Add/remove property and style values based on the number of new points.
+        with self.events.blocker_all(), self._border.events.blocker_all(), self._face.events.blocker_all():
```

Comment:
I feel like a lot of this logic could move to `_BasePoints`, with the derived classes calling super and then doing some minor class-specific things?